### ISSUE TITLE
Borg backup with USB Backup plugin

### ISSUE THREAD

**Issue by auanasgheps** [openmediavault/openmediavault#1734]
**Is your feature request related to a problem? Please describe.**
The USB backup plugin is great, and works very well. Rsync is a reliable tool, but has its limitations.

Me and other OMV users have been in the scenario of needing a more advanced backup solution. For example, Borg produces encrypted backups without the hassle of dealing with disk/partition encryption. This is ideal when storing the USB Drive "somewhere". Borg has also other advantages, but you get the point.

**Describe the solution you'd like**
Implement Borg backup as an option in the USB Backup plugin.

I am not a developer, but I believe the required work is fairly small. Borg backup has already an excellent interface via its own plugin, the work would be limited to trigger one or more Borg backup configuration(s) when the USB Backup has to start, instead of the rsync jobs.

**Describe alternatives you've considered**
I've considered doing this by myself with a script, but it's not that easy. That's why I've been using the USB Backup plugin since years.

**Additional context**
Happy to provide more information if you need.

**Comment by ryecoaaron** [comment]
Yes but it doesn't offer the ability to plug in a drive, backup to it, and then unmount.  I think I will look to see if it would be easier to add this to the borg plugin or the usbbackup plugin.

**Comment by auanasgheps** [comment]
Thanks guys for the fast response.

Using one of

**Comment by auanasgheps** [comment]
Thanks guys for the fast response.
Either implementation will do the job!

**Comment by NiklasTheis** [comment]
> Yes but it doesn't offer the ability to plug in a drive, backup to it, and then unmount. I think I will look to see if it would be easier to add this to the borg plugin or the usbbackup plugin.

Hi @ryecoaaron , is there any chance you cold give us some update on this request?
I would be very happy to be able to use this feature.

A question to all other people: do you have some kind of reliable and usable workaround for this case, which can be used until it is fully implemented?

**Comment by ryecoaaron** [comment]
> is there any chance you could give us some update on this request?

I started working on it but haven't had much time to work on it.  It is a bigger change than I thought it would be.

>  do you have some kind of reliable and usable workaround for this case, which can be used until it is fully implemented?

Either leave the drive plugged in all the time and use the borgbackup plugin OR use the usbbackup plugin as it is OR something manual.
